### DISCUSSION TITLE
`SIGSEGV` in  `withTaskCancellationHandler(operation:onCancel:isolation:)`

### DISCUSSION THREAD

**Discussion by setoelkahfi** [pointfreeco/swift-composable-architecture#3482] [General]
### Description

Hello, we have started migrating our codebase to Swift 6 language mode and fully migrated our TCA to be concurrency-ready for quite some time now. We updated our Xcode 16 and started seeing this crash on tvOS 18:

```
Project
+0xd4f76c
$ss27withTaskCancellationHandler9operation8onCancel9isolationxxyYaKXE_yyYbXEScA_pSgYitYaKlFTwbyt_Tg5 (<compiler-generated>)
In App

Project
+0xd4f76c
closure in RootStore.send (RootStore.swift:135)
In App

Called from
libswift_Concurrency
+0x042150
swift::runJobInEstablishedExecutorContext
```
Any clue?

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [ ] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

No crash.

### The Composable Architecture version information

1.15

### Destination operating system

tvOS 18 and up.

### Xcode version information

Xcode 16

### Swift Compiler version information

```shell
swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @setoelkahfi, unfortunately there is not enough information here to be of any help. At the very least a more complete stack trace would help, but ideally you would be able to provide a minimal project that demonstrates the crash. Can you try to do that as it helps us a lot and almost always leads to a resolution of the problem.

Since it isn't clear that this is an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!

**Comment by jshier** [comment]
There are multiple crashes in the global `with*` functions right now, depending on OS and Xcode version. https://forums.swift.org/t/global-with-concurrency-functions-crash-on-catalyst-designed-for-ipad-on-macos/74823/1 Update to Xcode 16.1 and some may be fixed.

**Comment by setoelkahfi** [comment]
Updating the project to Xcode 16.1 fixes the issue.
